docs/design/2022-03-09-optimize-gc-for-internal-transaction.md

```diff
@@ -0,0 +1,139 @@
+# Optimize GC Advance For Internal Transactions
+
+- Author(s): [Michael](https://github.com/TonsnakeLin) 
+- Tracking Issue: https://github.com/pingcap/tidb/issues/32725
+
+## Introduction
+
+This document describes the design of the feature "optimize gc advance for internal transaction", which adds the startTS(start timestamp) of all internal transactions to the gc(garbage collect) safepoint calculation process.
+
+## Motivation or Background
+
+TiDB advances gc safe point every `tidb_gc_run_interval` time with the value `now - tidb_gc_life_time`. If there is a long time transaction from user client lives more than `tidb_gc_life_time`, the safepoint can't be advanced until the transaction is finished or lives over 24 hours. This mechanism ensures the continuous advancement of gc safepoint and ensures that the data active transactions need to access will not be cleared.
```

Comment:
```suggestion
TiDB advances gc safepoint every `tidb_gc_run_interval` time with the value `now - tidb_gc_life_time`. If there is a long time transaction from user client lives more than `tidb_gc_life_time`, the safepoint can't be advanced until the transaction is finished or lives over 24 hours. This mechanism ensures the continuous advancement of gc safepoint and ensures that the data active transactions need to access will not be cleared.
```

There are still some "safe point" not unified. I have no idea which is better though, let's adopt @you06 's suggestion here.